### PR TITLE
Rattlemebones! Spessmens are no longer made of flubber.

### PR DESCRIPTION
**Pull request by Kevak**
**Testmerge/Review Ready.**

A total of:
44 Flypeople
44 Slimepeople
13 Jellypeople
23 Humans
4 Abductors
8 Golems
Have been harmed in the making of this so far.

-Bones are now a thing.
-Splints are a thing.
-Bonesetters are a thing
-Bonegel is a thing.
-Bonebreaking is damage threshold based. Not RNG based. This is not baymed. This is paramed.
-Limbs break easier than skulls and ribs.
-Walking around with broken, unsplinted limbs will damage you every so often.
-Bones can be mended in surgery. Placeholder surgery is scapal>retractor>hemostat>cautery. Will be changed later.
-Slimes do not and will not have bones.
-Boneless races now have a special surgery called Boneless Surgery. This lets the blobs of flubber and skittles skip through the sawing and closing up phases.
-Currently podpeople/plasmamen/skeletons do not have bones, this will change when I get a thing working. Ideally skeletons will be able to _Tear Out The Bones Of The Still Living._ As well as a less violent way of repairing broken bones.
-Monkeys have bones.
-Monkeys can have their bones repaired.
-Sharp objects do not break bones as well as blunt objects.
-Objects that deal more damage break bones better than lower force objects.
-Burn damage does not break bones.
-Splints can be found in mining venders, scattered around the station, and medbay. Look where medkits usually are.
-Currently Simplemobs that do not fire projectiles are exempted from bonebreak procs. _I Think._  Tested a bit but not extensively. As such, miners probably do not currently need to fear breaking their bones fighting angry fauna. This will change later. Mobs that fire projectiles still break bones if they do brute damage. It is 'possible' that they do break bones. I have not tested fully.
-As a side note, Bone Splints are available in mining vendors for miners. Target broken limb and clickself to apply.
-Golems have their own organ manip now. Saw>Retract>Saw>Retract>Manip>Bonegel>Boneset>Bonegel>Cautery. Will refluff later. Headache now.

Golems have their own surgeries as well.

:cl: Kevak/Neri
add: Spessmen now have Bones! 
balance: These can break!
balance: All expanded surgeries are visible on the surgery consoles. Use them if you need help!
add: Splints! These can be found scattered around stations.
add: Golems have their own surgery steps now. Saw>Retract>Saw>Retract>Special Surgery Step>Bonegel>Boneset>Bonegel>Cautery.
add: Surface surgeries, ala implant removal for golems have their own surgery path as well. Saw>retract>Special>Bonegel>Setter>Gel>Cautery.
add: Bonesetters and Bonegel! These can be found in appropriate surgery locations.
add: Surgery for bonerepair! scapal>retractor>hemostat>bonegel>bonesetter>bonegel>cautery
balance: Bones are a thing! This is obviously a major balance change. Some races do not have bones!
balance: Boneless species now have a special surgery called Boneless Organ Manipulation! scapal>retractor>hemostat>scapal>manip>cautery
balance: Some surgeries have been expanded. For anything requiring a saw. do a bonegel>bonesetter>bonegel before you cauterize.
balance: Boned organ manipulation is expanded! scapal>retractor>saw>hemostat>scapal>manip>retractor>bonegel>bonesetter>bonegel>cautery
fix: Slightly cleaned up health analyzer code.
fix: Surgery code is totally unfucked! No more stray numbers or 1/0's!
fix: Surgery code is now more modular!
wip: Bones are a WIP, they are currently safe to testmerge for stress testing and player feedback.
imageadd: Onsprite splint icons
imageadd: Splintstack icons
imageadd: Bonegel icons
imageadd: Bonesetter icons
experiment: Bones!
/:cl:

Why: Makes combat more interesting, Makes people be more tactical with their engagements. Stops people from trying to bullet sponge. Gives medbay something to do midround. And more.

Also make surgeries more sensible fluffwise.

Podpeople/plaspeople/skeletons will get their own PR for additional bone related features later. Slimepeople will be getting a debuff in that as well. Boneless is a very powerful thing.

### CODE REVIEW

**Comment by Kevak** [comment]
I don't have a damn clue how to add labels. Someone add Experimental, WIP, and Do Not Merge to this.

**Comment by WewLads** [comment]
where's my bone hurting juice

**Comment by Kevak** [comment]
Bone hurting juice is not a planned feature. Maybe when I give skeletons the ability to steal bones I'll add a regent that dissolves bones.

**Comment by pantherpusher** [comment]
Maybe in the future make bones obtainable from skeletonized remains

**Comment by Kevak** [comment]
Bones aren't currently an item Panther. They're a variable of a given limb. I'd need sprites if I wanted to make bones harvestable as an object you can carry around.

**Comment by silicons** [comment]
Uh why 126k lines you used mapmerge right?

**Comment by silicons** [comment]
TRUE FALSE defines

**Comment by praisenarsie** [comment]
>baymed

**Comment by Kevak** [comment]
I did indeed use map merge! It fucked anyways. I'll be updating based on the new remap anyways and just bypass that fuck.

**Comment by Kevak** [comment]
Unfucked box.

**Comment by Kevak** [comment]
..Uh. Fucking what? Boxstation specifically doesn't like being mapped.

**Comment by Kevak** [comment]
Aaaand ready for testmerge again. Will hold off next set of features till post test.

**Comment by Kevak** [comment]
Aaand borked surgery is now fixed. Also golems have surgery now for organ manip.

**Comment by Kevak** [comment]
Aand bonerepair unfucked. Golemsurgery will be disabled till I get to it.

**Comment by silicons** [comment]
whatever ping me when its ready

**Comment by Kevak** [comment]
I'll let ye know when it's fully done and ready for review. Currently I'd like to unfuck surgery further followed by finishing setting up specialized boneless surgeries for all types that matter and specialized golem surgeries. There's a lotta 1/0's instead of TRUE/FALSE's in surgery code I wanna fix up too.

And then that's more or less it. I can handle the atypical species bones stuff in a followup pr and other features in said followup.

**Comment by Kevak** [comment]
Looks like liveserver is still having simplemob issues. Gonna test locally to see if the fix is working or not. If it is, then it's an issue of 'Jay never merged the fix.'

**Comment by Kevak** [comment]
Fuck box station. Buggiest piece of shit to merge.

**Comment by Kevak** [comment]
Hopefully the Travis is fine with this. It /Should/ be. if it is then still don't testmerge cause I need to readd splints n shit to box. Fuck box. Shitty buggy map to merge. All my merge conflicts come from it. All of them.

**Comment by Kevak** [comment]
Woo! Travis worked! Now to map and then sleep

**Comment by Kevak** [comment]
@kevinz000 Review ready.

File: code/modules/surgery/limb_augmentation.dm [hunk 1]
```diff
@@ -15,19 +17,21 @@
...
 	time = 32
-	var/obj/item/bodypart/L = null // L because "limb"
+	var/obj/item/bodypart/L // L because "limb"
 
 
 /datum/surgery_step/add_limb/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
 	var/obj/item/bodypart/aug = tool
 	if(aug.status != BODYPART_ROBOTIC)
 		to_chat(user, "<span class='warning'>that's not an augment silly!</span>")
-		return -1
```
Comment (silicons) [inline code/modules/surgery/limb_augmentation.dm:27]: hope you know how surgery steps work like down to the finest detail, -1 is not 0.

**Comment by silicons** [comment]
i see no code issues however the final decision is not mine.

**Comment by cebutris** [comment]
okay but foxes are hollow so why do they have bones.

👍 good PR

File: code/modules/surgery/limb_augmentation.dm (hunk 1 above)
Comment (Kevak) [reply code/modules/surgery/limb_augmentation.dm:27]: I was under the assumption that -1 was 0, what does it parse as if not?

**Comment by LetterJay** [comment]
#3278

**Comment by Kevak** [comment]
Do not merge #3278 Jay. I already took everything of value from it and rolled it into this PR. 3278 will only break everything.

**Comment by LetterJay** [comment]
Is this ready for merge, @Kevak?

**Comment by Kevak** [comment]
No. You decided to merge 3278. Which broke everything. So I need to unfuck it. Again.

**Comment by Kevak** [comment]
Wait scratch that. Looks like you did not merge it now that I just looked. I do have a bug I need to fix though. So lemme do that. Will ping when ready.

**Comment by LetterJay** [comment]
I didn't merge it, I closed it. I'm adding a DNM then. Tage this with [Ready] when it's merge ready.

**Comment by Kevak** [comment]
Yah, add a DNM. I'll ping ye when it's ready, unable to add tags.

**Comment by Kevak** [comment]
Re: Progress. Been fixing runtimes in local/squashing bugs. Keeping repository untouched until done so I can roll back as needed.

**Comment by LetterJay** [comment]
Stale

**Comment by davethwave** [comment]
Ouch

**Pull request closed without merging**